### PR TITLE
Fix: version detection in abi diff workflow

### PR DESCRIPTION
**Pull request by Clueliss**
Fixes:
- private conan repository was missing
- The abi diff workflow previously did not correctly regenerate the version header for the base branch
- it was also not able to cope with cmake files that require a build step (instead of just the configuration step) to generate the version header (this is unusual, but we do it)

TODO:
- [x] point subworksflows to main

### CODE REVIEW

File: .github/workflows/abi-diff.yml [hunk 1]
```diff
@@ -87,57 +96,72 @@ jobs:
...
+          # this is a workaround, if I put the github expansions directly into
+          # the line below these two it doesn't work
+          base_branch="${{ inputs.base-branch }}"
+          search_path="${{ inputs.search-path }}"
+          
+          diff_lines=$(git diff "remotes/origin/$base_branch" -- $search_path | wc -l)
+          echo "Diff has $diff_lines lines"
+          echo "nlines=$diff_lines" >> "$GITHUB_OUTPUT"
+
+      # TODO point to main
```
Comment (Clueliss) [inline .github/workflows/abi-diff.yml:129]: The stupid thing with workflows is that you always have to specify a branch, so I basically need to break it by changing it to main and then immediately merge this PR into main to unbreak it. Idk if there is a better way to do this

Comment (bigerl) [reply .github/workflows/abi-diff.yml:129]: I think there is no better way. It's self-referential hell :)

File: .github/workflows/abi-diff.yml [hunk 2]
```diff
@@ -87,57 +96,72 @@ jobs:
...
           echo "abi_version=$abi_version" >> "$GITHUB_OUTPUT"
 
+      - name: Switch to base branch
+        run: |
+          rm -rf build_dir
+          git switch ${{ inputs.base-branch }}
+
+      # TODO point to main
+      - name: Regenerate version file for base branch
+        uses: dice-group/cpp-conan-release-reusable-workflow/.github/actions/generate_version_file@fix/abi-diff-private-conan
```
Comment (bigerl) [inline .github/workflows/abi-diff.yml:154]: change to main before merge here as well.

**Review by bigerl — APPROVED** [review]
Thx for the PR. The changes look good.

**Pull request merged**